src/test/java/com/example/petbuddybackend/controller/CaretakerControllerTest.java

```diff
@@ -72,4 +81,44 @@ void getCaretakers_shouldReturnFilteredResults() throws Exception {
                 .andExpect(jsonPath("$.content[0].accountData.name").value("John Doe"))
                 .andExpect(jsonPath("$.content[1].accountData.name").value("Jane Doe"));
     }
+
+    @ParameterizedTest
+    @MethodSource("provideRatingData")
+    @WithMockUser(username = "client")
```

Comment:
mamy takiego uzytkownika?

Reply:
Pytasz o `@WithMockUser(username = "client")`? To jest potrzebne do testu żeby ustawić `Principal principal` w kontrolerze